chameleonultragui/lib/bridge/chameleon.dart

```diff
@@ -343,12 +343,37 @@ class ChameleonCom {
   Future<int> getFirmwareVersion() async {
     var resp = await sendCmdSync(ChameleonCommand.getAppVersion, 0x00);
     if (resp!.data.length != 2) throw ("Invalid data length");
-    return (resp.data[1] << 8) | resp.data[0];
+    final result = (resp.data[1] << 8) | resp.data[0];
+    log.d("ChamelonCom.getFirmwareVersion: $result");
+    return result;
   }
 
   Future<String> getDeviceChipID() async {
     var resp = await sendCmdSync(ChameleonCommand.getDeviceChipID, 0x00);
-    return bytesToHex(resp!.data);
+    final result = bytesToHex(resp!.data);
+    log.d("ChamelonCom.getDeviceChipID: $result");
+    return result;
+  }
+
+  Future<bool> detectChameleonUltra() async {
+    // Check if device is a ChameleonUltra by trying to switch to reader mode,
```

Comment:
Better to implement in FW... but nobody forbids to do later

Reply:
Agree, its on my todo. Personally I feel it would be best if ChameleonLite's had a unique product id as they are just not the same product as Ultra's. Should be a quick change

Reply:
You can use DATA_CMD_WRITE_EM410X_TO_T5577 without any data, will throw STATUS_PAR_ERR if ultra and won't exit emulator mode